lib/fs.js

```diff
@@ -721,28 +721,28 @@ function fsyncSync(fd) {
 }
 
 function mkdir(path, options, callback) {
-  const defaults = { parent: false, mode: 0o777 };
   if (typeof options === 'function') {
     callback = options;
-    options = defaults;
+    options = {};
+  }
+  if (typeof options === 'number' || typeof options === 'string') {
```

Comment:
Micro-optimization, but you can link these blocks with `else` to avoid evaluating these 2 type-checks if `options` was a function.

Reply:
Probably needs to be elseif